### PR TITLE
handle leading 0 in restricted district id

### LINKED ISSUE #1148: Error: You do not have permission to view this content.
I have found the login that this is tied to - and the user does have access to district 03. The user was able to successfully log in again this morning without triggering this auth error. Trying to figure out what happened on Thursday...

Sentry Issue: [RECIDIVIZ-PULSE-DASHBOARD-AD](https://sentry.io/organizations/recidiviz-inc/issues/2474990280/?referrer=github_integration)

```
Error: You do not have permission to view this content.
  at e.value (lantern/LanternStore/UserRestrictionsStore.ts:69:25)
  at _t (/static/js/2.6dafe568.chunk.js:2:14138)
  at e.verifyUserRestrictions (/static/js/2.6dafe568.chunk.js:2:13673)
  at None (lantern/LanternStore/index.ts:74:36)
  at u (/static/js/2.6dafe568.chunk.js:2:27167)
...
(15 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by jovergaag**
## Description of the change

We had an issue last week where MO had entered district ids for some users that were missing the leading 0. So the id was "3" instead of "03". This PR converts any strings that are integers to integers while verifying the id to ensure that a leading 0 does not cause an error.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #1148

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [x] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by daschi — COMMENTED** [review]
Nice validation! Curious if we also want to find and use the supervision location ID from the verified locations list here? If this happens again, the user won't see an auth error, but they also won't see their district's data because it'll be filtering by the wrong ID.

**Comment by jovergaag** [comment]
@daschi Great catch! I changed it so that the `UserRestrictionsStore` is using the normalized id for `allowedSupervisionLocationIds`

**Review by Catacola — COMMENTED** [review]
Everything looks good, but I just have an itch about some of the repeated patterns. If there isn't a clean solution, we can go with this.

File: src/lantern/LanternStore/UserRestrictionsStore.ts [hunk 1]
```diff
@@ -50,26 +51,33 @@ export default class UserRestrictionsStore {
...
-    const unverifiedLocations = this.allowedSupervisionLocationIds.filter(
-      (supervisionLocationId) => {
-        return !this.rootStore.districtsStore.districtIds.includes(
-          supervisionLocationId
-        );
+    const normalizedIds = this.rootStore.districtsStore.districtIds.filter(
+      (district) => {
+        return this.rootStore.userStore.allowedSupervisionLocationIds
+          .map((d) => safeToInt(d))
+          .includes(safeToInt(district));
```
[the remark this answers is not in the document]
Comment (jovergaag) [reply src/lantern/LanternStore/UserRestrictionsStore.ts]: Good call!

File: src/lantern/LanternStore/UserRestrictionsStore.ts [hunk 2]
```diff
@@ -20,6 +20,13 @@ import { ERROR_MESSAGES } from "../../constants/errorMessages";
 import type LanternStore from ".";
 import { US_MO } from "../../RootStore/TenantStore/lanternTenants";
 import { CHARTS } from "./DataStore/RevocationsChartStore";
+import { safeToInt } from "../../utils";
+
+const typeSafeIncludes = (arrayA: string[], arrayB: string[]): string[] => {
+  return arrayA.filter((a: string) => {
+    return arrayB.map((b: string) => safeToInt(b)).includes(safeToInt(a));
+  });
+};
```
Comment (Catacola) [inline src/lantern/LanternStore/UserRestrictionsStore.ts:29]: Thanks! I think the code below is now much easier to follow.

**Review by Catacola — APPROVED** [review]

**Pull request merged**